nevergrad/functions/base.py

```diff
@@ -204,6 +204,36 @@ def update_leaderboard(identifier: str, loss: float, array: np.ndarray, verbose:
         pass  # better avoir bugs for this
 
 
+class ArrayExperimentFunction(ExperimentFunction):
+    """Adds a "symmetry" parameter, which allows the creation of many symmetries of a given function."""
+
+    def symmetrized_function(self: AEF, x: tp.Any):
+        assert isinstance(x, np.ndarray), "symmetry != 0 works only when the input is an array."
+        assert len(x.shape) == 1, "only one-dimensional arrays for now."
+        y = x
+        symmetry: int = self._symmetry  # type: ignore
+        for i in range(len(y)):
+            if symmetry % 2 == 1:
+                y[i] = -x[i]  # We should rather symmetrize w.r.t the center of Parameter. TODO
+            symmetry = symmetry // 2
+        return self._inner_function(y)  # type: ignore
+
+    def __init__(self: AEF, function: tp.Callable[..., tp.Loss], parametrization: p.Parameter, symmetry: int = 0) -> None:
+        """ Same parameters as ExperimentFunction, plus "symmetry".
+        symmetry: an int, 0 by default.
+        if not zero, a symmetrization is applied to the input; each of the 2^d possible values
+        for symmetry % 2^d gives one different function.
```

Comment:
maybe symmetry >= 2d should not be allowed?

Reply:
Done.

Reply:
Maybe the creation of all symmetries, of a subset of all symmetries, should be possible... for later.